common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -582,7 +583,14 @@ private boolean runShutdownHooks() {
         gracefulShutdownTimeout = unit.toNanos(timeout);
 
         if (oldState == ST_NOT_STARTED) {
-            doStartThread();
+            try {
+                doStartThread();
+            } catch (RejectedExecutionException cause) {
```

Comment:
`j.u.c.RejectedExecutionHandler` is not limited to throwing REEs, why not `Throwable`?

Reply:
We try to not introduce more Throwable catches if we rethrow these. Maybe `Exception` ?

Reply:
Sounds good.

Reply:
@rvansa PTAL again just pushed